tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_multiple_snapshots.py

```diff
@@ -0,0 +1,414 @@
+                time.sleep(30)
+
+                # Check if the maximum duration has been reached
+                elapsed_time = time.time() - start_time
+                if elapsed_time >= max_duration_seconds:
+                    log.info("Maximum duration reached. Breaking out of the loop.")
+                    break
+            else:
+                break
+
+        fs_mirroring_utils.validate_snapshot_sync_status(
+            cephfs_mirror_node,
+            source_fs,
+            snap2,
+            fsid,
+            asok_files,
+            filesystem_id,
+            peer_uuid_new,
+        )
+        peer_uuid = peer_uuid_new
+        out, rc = fs_mirroring_utils.list_and_verify_remote_snapshots_and_data_checksum(
+            target_clients[0],
+            "client.admin",
+            subvol2_path,
+            source_clients[0],
+            fuse_mounting_dir_1,
+            target_fs,
+        )
+        if not out:
+            raise CommandFailed(f"{rc}")
+        log.info(
+            f"{snap2} was synced with out any errors even when mirror node bootstrap token is removed and added back"
+        )
+        source_clients[0].exec_command(
+            sudo=True,
+            cmd=f"mkdir -p {kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir}/{mounting_dir}_1;"
+            f"python3 /home/cephuser/smallfile/smallfile_cli.py "
+            f"--operation create --threads 1 --file-size 1 "
+            f"--files 5 --files-per-dir 1 --dirs-per-dir 2 --top "
+            f"{kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir}/{mounting_dir}_1",
+        )
+        source_clients[0].exec_command(
+            sudo=True,
+            cmd=f"mkdir -p {kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir}_1;"
+            f"python3 /home/cephuser/smallfile/smallfile_cli.py "
+            f"--operation create --threads 1 --file-size 1 "
+            f"--files 5 --files-per-dir 1 --dirs-per-dir 2 --top "
+            f"{kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir}_1",
+        )
+
+        source_clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume snapshot create {source_fs} subvol_1 {snap1} --group_name subvolgroup_1",
+        )
+
+        fs_mirroring_utils.validate_snapshot_sync_status(
+            cephfs_mirror_node,
+            source_fs,
+            f"{snap1}",
+            fsid,
+            asok_files,
+            filesystem_id,
+            peer_uuid,
+        )
+        time.sleep(60)
+        out, rc = fs_mirroring_utils.list_and_verify_remote_snapshots_and_data_checksum(
+            target_clients[0],
+            "client.admin",
+            subvol1_path,
+            source_clients[0],
+            kernel_mounting_dir_1,
+            target_fs,
+        )
+        if not out:
+            raise CommandFailed(f"{rc}")
+        log.info(
+            f"{snap1} was synced with out any errors even when mds nodes disconnected on source cluster"
```

Comment:
Incorrect error message? "{snap1} was synced with out any errors even when  {snap1} was recreated on source cluster...." should be msg instead?